build/room_ensurer/room_ensurer.py

```diff
@@ -270,6 +276,36 @@ def _create_server_user_power_levels(self) -> Dict[str, Any]:
 
         return server_admin_power_levels
 
+    def _ensure_admin_power_levels(self, room_info: RoomInfo) -> None:
+        log.info(f"Ensuring power levels for {room_info.aliases}")
+        api = self._apis[self._own_server_name]
+        own_user = f"@{self._username}:{self._own_server_name}"
+        supposed_power_levels = self._create_server_user_power_levels()
+
+        try:
+            current_power_levels = api.get_room_state_type(
+                room_info.room_id, "m.room.power_levels", ""
+            )
+        except MatrixError:
+            log.debug("Could not fetch power levels", room_aliases=room_info.aliases)
+            return
+
+        if own_user not in current_power_levels["users"]:
+            log.warning(f"{own_user} has not been granted administrative power levels yet. Doing nothing.")
+            return
+
+        # the supposed power level dict could be just a subset of the current
+        # because providers who left cannot be removed from other admins
+        if set(supposed_power_levels["users"].keys()).issubset(set(current_power_levels["users"].keys())):
```

Comment:
Should we check the contents of the power levels here as well?
Related to the point above.

Reply:
The check itself is supposed to integrate old admin users which are not on the list anymore. The reason for this is that AFAIK one can only change the power levels of someone who has lower power levels. Therefore the list can only increase. 

maybe swapping the merge_dict arguments will help?

Reply:
Nah I think it's fine for now.